packages/jest-transform/src/types.ts

```diff
@@ -52,21 +52,70 @@ export interface CacheKeyOptions extends TransformOptions {
   rootDir: string;
 }
 
-export interface Transformer {
+interface SyncTransFormer {
   canInstrument?: boolean;
-  createTransformer?: (options?: any) => Transformer;
+  createTransformer?: (options?: any) => SyncTransFormer;
 
   getCacheKey?: (
-    fileData: string,
+    fileDate: string,
     filePath: Config.Path,
     configStr: string,
     options: CacheKeyOptions,
   ) => string;
 
+  getCacheKeyAsync?: (
+    fileDate: string,
+    filePath: Config.Path,
+    configStr: string,
+    options: CacheKeyOptions,
+  ) => Promise<string>;
+
   process: (
     sourceText: string,
     sourcePath: Config.Path,
     config: Config.ProjectConfig,
     options?: TransformOptions,
   ) => TransformedSource;
+
+  processAsync?: (
+    sourceText: string,
+    sourcePath: Config.Path,
+    config: Config.ProjectConfig,
+    options?: TransformOptions,
+  ) => Promise<TransformedSource>;
 }
+
+interface AsyncTransformer {
+  canInstrument?: boolean;
+  createTransformer?: (options?: any) => AsyncTransformer;
+
+  getCacheKey?: (
+    fileDate: string,
+    filePath: Config.Path,
+    configStr: string,
+    options: CacheKeyOptions,
+  ) => string;
+
+  getCacheKeyAsync?: (
```

Comment:
Not optional

Reply:
But originally, a `Transformer` might not have `getCacheKey` at all?

Reply:
oh? in that case it's optional 😀